### PR TITLE
explicitely support APs defined by CIDR/IP for Unifi web-auth

### LINKED ISSUE #6686: Unifi: deauthentication doesn't work anymore if AP are declared by IP address
**Describe the bug**
It looks like since #6134, deauthentication of a user doesn't work if it is connected on a Unifi access point which is declared by IP address (not by MAC address).

An AP declared by IP address will not pass `if` test:
https://github.com/inverse-inc/packetfence/blob/758bce50d9fb12c7d538745452507e16448b9d93/lib/pf/Switch/Ubiquiti/Unifi.pm#L272-L277

**To Reproduce**
1. Declared a Unifi AP by IP address
2. Authenticate on captive portal using Web Auth
3. Check logs in `packetfence.log`:

```
│Nov  5 08:58:39 pf01 pfqueue[132890]: pfqueue(132890) INFO: [mac:4c:32:75:70:0e:19] Found site: Test (pf::Switch::Ubiquiti::Unifi::_deauthenticateMacWithHTTP)                                                         │
│Nov  5 08:58:39 pf01 pfqueue[132890]: pfqueue(132890) INFO: [mac:4c:32:75:70:0e:19] Deauth on 0 access points (pf::Switch::Ubiquiti::Unifi::_deauthenticateMacWithHTTP)     
```

**Expected behavior**
Deauthentication should also work on AP declared by IP address.

**Additional context**
In this setup, AP are declared using a CIDR.

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
explicitely support APs defined by CIDR/IP for Unifi web-auth

# Impacts
Ubiquiti Unifi module

# Issue
fixes #6686 

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [N/A] Document the feature
- [N/A] Add unit tests
- [N/A] Add acceptance tests (TestLink)

# NEWS file entries
## Bug Fixes
* Fix regression in the Unifi module for deauthentication of webauth clients when the APs are defined using an IP or CIDR in the configuration (#6686)
*

### CODE REVIEW

**Comment by julsemaan** [comment]
I've marked this as a +1 since it doesn't alter the behavior that was released in 11.1, it just adds another behavior that is opinionated on APs defined using IP or CIDR

File: lib/pf/Switch/Ubiquiti/Unifi.pm [hunk 1]
```diff
@@ -292,11 +310,8 @@ sub _deauthenticateMacWithHTTP {
...
-    $logger->info("Deauth on $count access points");
-
-    unless($response->is_success) {
-        $logger->error("Can't send request on the Unifi controller: ".$response->status_line);
-        return;
+    if($count > 0) {
+    	$logger->info("Deauth on $count access points");
     }
 
     $logger->info("Switched status on the Unifi controller using command $command");
```
Comment (nqb) [inline lib/pf/Switch/Ubiquiti/Unifi.pm]: This sentence could be wrong if your device has not been found and your access points are defined by AP in PacketFence config.

**Review by nqb — CHANGES_REQUESTED** [review]
Thanks for your detailled comments.

IMHO, it's a problem to have different behavior depending on how AP are declared into PacketFence configuration. 

If I correctly understand your code, in a case where AP are declared by IP addresses in PacketFence configuration and MAC address of device is not found on a Unifi site, we don't deauthenticate. However, in such scenario for AP declared by MAC addresses, we deauthenticate on all sites.

**Comment by julsemaan** [comment]
> Thanks for your detailled comments.
> 
> IMHO, it's a problem to have different behavior depending on how AP are declared into PacketFence configuration.
> 
> If I correctly understand your code, in a case where AP are declared by IP addresses in PacketFence configuration and MAC address of device is not found on a Unifi site, we don't deauthenticate. However, in such scenario for AP declared by MAC addresses, we deauthenticate on all sites.

That is correct, we could make it so we send it to all sites if we don't find the device but from what I could see this wouldn't be necessary on the recent Unifi versions since we're sure to find it if its authorized or connected. This means when granting access we'll find it (since the user is connected and on the portal), and when we're revoking access, the entry should be found in a site since its currently authorized.

My understanding is that the block to deauth on all sites would be targetted at old Unifi versions but its hard to tell at the moment. We could always add the block to deauth on all sites for IP/CIDR APs, it won't cost more than me taking the time to do a copy/paste.

**Comment by nqb** [comment]
> My understanding is that the block to deauth on all sites would be targetted at old Unifi versions but its hard to tell at the moment. We could always add the block to deauth on all sites for IP/CIDR APs, it won't cost more than me taking the time to do a copy/paste.

I agree.

**Comment by nqb** [comment]
@julsemaan, let me know if you plan to adjust this PR.

**Comment by julsemaan** [comment]
PR adjusted

File: lib/pf/Switch/Ubiquiti/Unifi.pm (hunk 1 above)
Comment (nqb) [reply lib/pf/Switch/Ubiquiti/Unifi.pm]: This is still a problem for me.

Comment (nqb) [reply lib/pf/Switch/Ubiquiti/Unifi.pm]: I misunderstand logic. In fact, this sentence will never appear in logs if AP are defined by IP addresses.

**Pull request merged**